File: examples/redis/topologies/wordcount-redis.clj
(ns wordcount-redis
  (:use     [streamparse.specs])
  (:gen-class))

(defn wordcount-redis [options]
   [
    ;; spout configuration
    {"word-spout" (python-spout-spec
          options
          "wordcount_redis.WordSpout"
          ["word"]
          )
    }
    ;; bolt configuration
    {"count-bolt" (python-bolt-spec
          options
          ;; shuffle grouping since data sent to Redis
          {"word-spout" :shuffle}
          "wordcount_redis.WordCountBolt"
          ["word" "count"]
          :p 4
          )
    }
  ]
)

File: examples/redis/topologies/wordcount_redis.py
"""
Word count topology (in Redis)
"""

from src.bolts import RedisWordCountBolt
from src.spouts import WordSpout
from streamparse import Topology


class WordCount(Topology):
    word_spout = WordSpout.spec()
    count_bolt = RedisWordCountBolt.spec(inputs=[word_spout], par=4)
